xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/transactions/Payment.java

```diff
@@ -41,14 +44,32 @@ default PaymentFlags flags() {
   }
 
   /**
-   * The amount of currency to deliver. If the {@link Flags.PaymentFlags#tfPartialPayment()} flag is set, deliver
-   * up to this amount instead.
+   * The amount of currency to deliver. If the {@link Flags.PaymentFlags#tfPartialPayment()} flag is set, deliver up to
+   * this amount instead.
    *
    * @return A {@link CurrencyAmount} representing the amount of a specified currency to deliver.
    */
   @JsonProperty("Amount")
   CurrencyAmount amount();
 
+  /**
+   * A typed variant of {@link #amount()} which produces a value in XRP Drops if that result is an instance of {@link
+   * XrpCurrencyAmount}.
+   *
+   * @return The amount of this payment if this payment is denominated in XRP (i.e., if the result of calling {@link
+   *   #amount()} is an instance of {@link XrpCurrencyAmount}); otherwise, returns {@link Optional#empty()} (e.g., if
+   *   this payment is denominated in an Issued Currency).
+   */
+  @Derived
+  @JsonIgnore
+  default Optional<XrpCurrencyAmount> amountAsXrp() {
```

Comment:
I liked your idea of providing a handler on `CurrencyAmount`.  The handler could work in tandem with an `amountAsXrp` method like this:
```java
payment.amount().handle(
  xrpCurrencyAmount -> xrpCurrencyAmount.toXrp(),
  issuedCurrencyAmount -> issuedCurrencyAmount
)
```

There are a few transactions that have an `amount` field that could be an `XrpCurrencyAmount` or an `IssuedCurrencyAmount`, so I would prefer having this pulled up into `CurrencyAmount` or `XrpCurrencyAmount`

Reply:
Agreed. See latest commit here and let me know what you think?
https://github.com/XRPLF/xrpl4j/pull/68/commits/1dcd61f54b98b21a97053d05963a584efe7dc75d